cypress/e2e/models/migration/dynamicreports/issue.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    click,
+    clickByText,
+    clickWithinByText,
+    getUrl,
+    inputText,
+    selectFilter,
+    selectItemsPerPage,
+    selectUserPerspective,
+} from "../../../../utils/utils";
+import { button, migration, SEC } from "../../../types/constants";
+import { navMenu } from "../../../views/menu.view";
+import { searchButton } from "../../../views/common.view";
+import {
+    appFilterName,
+    bsFilterName,
+    categoryFilterName,
+    sourceFilterName,
+    tagFilterName,
+    targetFilterName,
+} from "../../../views/issue.view";
+
+export class Issue {
+    /** Contains URL of issues web page */
+    static fullUrl = Cypress.env("tackleUrl") + "/issues";
+
+    public static openList(itemsPerPage = 100, forceReload = false): void {
+        if (forceReload) {
+            cy.visit(Issue.fullUrl);
+        }
+        if (!getUrl().includes(Issue.fullUrl)) {
+            selectUserPerspective(migration);
+        }
+        clickByText(navMenu, "Issues");
+        cy.wait(2 * SEC);
+        selectItemsPerPage(itemsPerPage);
+    }
+
+    public static filterByName(name: string) {
+        selectFilter("Application name");
+        inputText(appFilterName, name);
+        click(searchButton);
+    }
+
+    public static filterByBs(name: string) {
+        selectFilter("Business service");
+        click(bsFilterName);
+        clickWithinByText(bsFilterName, button, name);
+    }
+
+    public static filterByTag(names: string[]) {
+        selectFilter("Tags");
+        click(tagFilterName);
+        names.forEach((name) => {
+            clickWithinByText(tagFilterName, "span", name);
+        });
+    }
+    public static filterByCategory(name: string) {
+        selectFilter("Category");
+        inputText(categoryFilterName, name);
+        click(searchButton);
+    }
+
+    public static filterBySource(name: string) {
+        selectFilter("Source");
+        inputText(sourceFilterName, name);
+        click(searchButton);
+    }
+    public static filterByTarget(name: string) {
+        selectFilter("Target");
+        inputText(categoryFilterName, name);
+        click(targetFilterName);
+    }
```

Comment:
These methods that are so similar can be done in one method, otherwise, we may have to add new methods for each new filter.

Idea:

```
public static filterIssue(filterName: string, filterInput: string, filterValue: string, filterButtonSelector: string) {
        selectFilter(filterName);
        inputText(filterInput, filterValue);
        click(filterButtonSelector);
    }
```

And even with some slightly changes it could also accept arrays (for multi-selectable filter like tags)

Reply:
Done, please take a look